test/e2e/multikueue/e2e_test.go

```diff
@@ -791,3 +872,30 @@ func getKuberayTestImage() string {
 	gomega.Expect(found).To(gomega.BeTrue())
 	return kuberayTestImage
 }
+
+func ensurePodWorkloadsRunning(pods *corev1.PodList, managerNs corev1.Namespace, multiKueueAc *kueue.AdmissionCheck) {
+	// Given the unpredictable nature of where the deployment pods run this function gathers the workload of a Pod first
+	// it then gets the Pod's assigned cluster from the admission check message and uses the appropriate client to ensure the Pod is running
+	ginkgo.By("Ensure Pod Workloads are created and Pods are Running on the worker cluster", func() {
+		for _, pod := range pods.Items { // We want to test that all deployment pods have workloads.
+			createdLeaderWorkload := &kueue.Workload{}
+			wlLookupKey := types.NamespacedName{Name: workloadpod.GetWorkloadNameForPod(pod.Name, pod.UID), Namespace: managerNs.Name}
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sManagerClient.Get(ctx, wlLookupKey, createdLeaderWorkload)).To(gomega.Succeed())
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+			// By checking the assigned cluster we can discern which client to use
+			admissionCheckMessage := workload.FindAdmissionCheck(createdLeaderWorkload.Status.AdmissionChecks, multiKueueAc.Name).Message
+			workerCluster := k8sWorker1Client
+			if strings.Contains(admissionCheckMessage, "worker2") {
+				workerCluster = k8sWorker2Client
+			}
```

Comment:
Right, this will work but relies on the assumption we have only 2 workers. I would suggest to introduce a helper function which gets the worker name in the utils, for re-usability.

Reply:
Ah, I see, yeah we can add the map which we initialize along with the clients.

Reply:
Could be a follow up - Wdyt about adding a label to the pods with the cluster it gets admitted into. That would help us filtering the relevant resources on the client side with label selector.